### ISSUE TITLE
Priority: PhpdocAnnotationWithoutDotFixer should run before PhpdocTypesFixer

### ISSUE THREAD

**Issue by dmvdbrugge** [FriendsOfPHP/PHP-CS-Fixer#3870]
In the example code, `phpdoc_types_order` and `phpdoc_annotation_without_dot` have been applied the first run, but they should not conflict, I see no reason why `phpdoc_types` couldn't have been applied too.

#### The PHP version you are using (`$ php -v`):
```
PHP 7.1.18 (cli) (built: May 25 2018 19:18:59) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.1.18, Copyright (c) 1999-2018, by Zend Technologies
```

#### PHP CS Fixer version you are using (`$ php-cs-fixer -V`):
`PHP CS Fixer 2.12.1 Long Journey by Fabien Potencier and Dariusz Ruminski`

#### The command you use to run PHP CS Fixer:
`vendor/bin/php-cs-fixer fix -v --using-cache=no`

#### The configuration file you are using, if any:
Basically the same as [dmvdbrugge/dynamic-components](https://github.com/dmvdbrugge/dynamic-components/blob/5dfba1d7ea9573b2f2d75468556c988019be541c/.php_cs.dist), except `explicit_string_variable`, `header_comment`, `no_alternative_syntax`, and `no_unset_on_property`

#### If applicable, please provide minimum samples of PHP code (as plain text, not screenshots):
 * before running PHP CS Fixer (no changes):
```php
/**
 * @return int[]|string[]|NULL[].
 */
```

* after first run:
```php
/**
 * @return int[]|NULL[]|string[]
 */
```

* after second run:
```php
/**
 * @return int[]|null[]|string[]
 */
```

* expected:
Immediately arrive at last step

Meta: this is case 48 of #3844

**Comment by kubawerlos** [comment]
Are you sure it's `LowercaseConstantsFixer`? It is not touching comments. Didn't you mean `PhpdocTypesFixer`?

**Comment by dmvdbrugge** [comment]
Makes more sense indeed. Didnt test te individual rules in thuis case :oops:

**Comment by kubawerlos** [comment]
However, the initial dot being too close to the type is the problem - it is treated as part of type, making it invalid - in [THE RegEx](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.12.1/src/DocBlock/Annotation.php#L30) - I'm not sure `phpdoc_annotation_without_dot` shall remove it in the first place, as it's job is to make sure "PHPDoc annotation descriptions should not be a sentence" and in this case it's not the description.

**Comment by dmvdbrugge** [comment]
However it does only check for "the last dot of the line" so it will remove it
